### PR TITLE
removed members should also be removed from account_origins

### PR DESCRIPTION
**Pull request by mwrock**
Signed-off-by: mwrock <matt@mattwrock.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-depot/src/server.rs [hunk 1]
```diff
@@ -473,17 +473,28 @@ pub fn origin_member_delete(req: &mut Request) -> IronResult<Response> {
...
         Err(err) => return Ok(render_net_error(&err)),
     }
-    request.set_account_name(account_name.to_string());
+    session_request.set_account_id(session.get_id());
+    origin_request.set_account_name(account_name.to_string());
+
+    if let Err(err) = route_message::<AccountOriginRemove, NetOk>(req, &session_request) {
+        error!("Error deleting origin_account from sessionsrv, {}", err);
+        return Ok(render_net_error(&err));
+    }
```
Comment (raskchanky) [inline components/builder-depot/src/server.rs]: I wonder if it'd be smart to only run the delete from the sessionsrv if the delete from the originsrv succeeded, or vice versa.  The point being, ensure that either both succeed or neither succeeds.  Right now, though it's unlikely, you could have the delete from sessionsrv succeed and the delete from originsrv fail, and now they're out of sync.

Thoughts?

Comment (mwrock) [reply components/builder-depot/src/server.rs]: i thought about this. I think its more important to have the sessionsrv delete succeed and originsrv fail. The reason being that if for some reason a partial deletion occured and only sessionsrv succeeded. At least the removed user would not be able to see the origin anymore.  The weirdneww would be that the user would still show up as a member of the origin, but a "reremoval" could be attempted. The opposite could be worse: the member could still see the origin but legit origin members would not see that user as a member.

Comment (raskchanky) [reply components/builder-depot/src/server.rs]: I think my point was more along the lines of only attempting the second delete if the first one succeeds, regardless of which one you do first.  We don't have a way of doing a distributed commit with this stuff, but at least that way, if the first one fails, you don't even try the second delete, and just abort completely.

Comment (mwrock) [reply components/builder-depot/src/server.rs]: oh that should already be setup. If the first fails it does `return Ok(render_net_error(&err))`

Comment (raskchanky) [reply components/builder-depot/src/server.rs]: Haha, no idea how I missed that.  👍

**Comment by mwrock** [comment]
related to #3513

File: components/builder-depot/src/server.rs (hunk 1 above)
Comment (reset) [reply components/builder-depot/src/server.rs]: @mwrock this is a good starting place and better than having nothing, but what we want to do is use the `async_job` feature of the db crate that @adamhjk added. When you have a database transaction that crosses a shard partition you need to use the retry logic in the server of record to ensure the other bits are cleaned up

With that said, I think it's ok to merge this in as is for now but we should add a comment and circle around back

**Comment by reset** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/284462079).

File: components/builder-depot/src/server.rs (hunk 1 above)
Comment (mwrock) [reply components/builder-depot/src/server.rs]: ok thanks. I'll look for a usage example

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/284462079).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**